src/coreclr/tools/aot/crossgen2/crossgen2.csproj

```diff
@@ -1,9 +1,63 @@
-<Project Sdk="Microsoft.NET.Sdk">
+<Project>
+
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
   <PropertyGroup>
-    <OutputPath>$(RuntimeBinDir)/crossgen2</OutputPath>
-    <!-- The default value for macOS is false -->
-    <UseAppHost>true</UseAppHost>
-    <AppHostRuntimeIdentifier>$(Crossgen2PackageRID)</AppHostRuntimeIdentifier>
+    <OutputPath>$(RuntimeBinDir)crossgen2</OutputPath>
+    <DisablePackageAssetsCache Condition="'$(DotnetBuildFromSource)' == 'true'">true</DisablePackageAssetsCache>
```

Comment:
Why is that setting required?

Reply:
Ah, I can probably remove that now. The problem was that we do two restores: one from `Build.proj`, and a second one here when we call Publish on `crossgen2`. There was a big problem with caching here.

I've now changed the RuntimeIdentifiers to include the source build RID in the initial restore, so even if it's cached, it should be OK. I'll remove that.